src/templates/doc.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { graphql } from 'gatsby'
+import { Node } from 'unist'
+import { getItemByPath } from '../utils/shared/sidebar'
+
+import SEO from '../components/SEO'
+
+import Documentation from '../components/Documentation'
+
+interface IDocPageProps {
+  data: {
+    page: {
+      htmlAst: Node
+    }
+  }
+  pageContext: {
+    slug: string
+    headings: []
+  }
+}
```

Comment:
Obviously there's some massive code duplication right now which codeclimate picks up on, but it's likely different templates' data fetching will require different interfaces and queries.

I would recommend we only worry about this kind of refactoring somewhere between the page being finished and merging to master.

Reply:
👍 let's not worry about CC at this stage, agreed